### PR TITLE
CB-11198 Skip android target sdk check

### PR DESCRIPTION
**Pull request by nikhilkh**

### CODE REVIEW

File: bin/lib/check_reqs.js [hunk 1]
```diff
@@ -254,18 +252,22 @@ module.exports.check_android_target = function(valid_target) {
...
 
         var androidCmd = module.exports.getAbsoluteAndroidCmd();
-        throw new CordovaError('Please install Android target: "' + valid_target + '".\n\n' +
+        var msg = 'Please install Android target: "' + valid_target + '".\n\n' +
             'Hint: Open the SDK manager by running: ' + androidCmd + '\n' +
             'You will require:\n' +
             '1. "SDK Platform" for ' + valid_target + '\n' +
             '2. "Android SDK Platform-tools (latest)\n' +
-            '3. "Android SDK Build-tools" (latest)');
+            '3. "Android SDK Build-tools" (latest)'
```
Comment (jasongin) [inline bin/lib/check_reqs.js]: Missing semicolon.

File: bin/lib/check_reqs.js [hunk 2]
```diff
@@ -238,13 +236,13 @@ module.exports.getAbsoluteAndroidCmd = function () {
     return cmd.replace(/(\s)/g, '\\$1');
 };
 
-module.exports.check_android_target = function(valid_target) {
+module.exports.check_android_target = function(originalError) {
```
Comment (jasongin) [inline bin/lib/check_reqs.js]: I don't see where any value for originalError is ever passed in by a caller.

File: bin/templates/cordova/lib/builders/GradleBuilder.js [hunk 3]
```diff
@@ -203,6 +203,11 @@ GradleBuilder.prototype.build = function(opts) {
         } else {
             process.stdout.write(stdio.stdout);
         }
+    }).catch(function (error) {
+        if (error.toString().indexOf('failed to find target with hash string') >= 0) {
+            return check_reqs.check_android_target(error);
```
Comment (jasongin) [inline bin/templates/cordova/lib/builders/GradleBuilder.js]: Would it be possible for check_android_target to actually find the target and then swallow this error?

**Comment by jasongin** [comment]
:+1:

File: bin/templates/cordova/lib/builders/AntBuilder.js [hunk 4]
```diff
@@ -107,7 +107,18 @@ AntBuilder.prototype.build = function(opts) {
     var args = this.getArgs(opts.buildType == 'debug' ? 'debug' : 'release', opts);
     return check_reqs.check_ant()
     .then(function() {
-        return spawn('ant', args, {stdio: 'inherit'});
+        return spawn('ant', args, {stdio: 'pipe'});
+    }).progress(function (stdio){
```
Comment (vladimir-kotikov) [inline bin/templates/cordova/lib/builders/AntBuilder.js]: Should we handle stdio of `ant build` manually here?

Comment (nikhilkh) [inline bin/templates/cordova/lib/builders/AntBuilder.js]: Can you elaborate? As far as I can see superspawn only returns `stderr` output only if you 'pipe'. Piping implies that someone has to still write this to process.stdout & process.stderr.

Comment (vladimir-kotikov) [inline bin/templates/cordova/lib/builders/AntBuilder.js]: My bad, I somehow ignored the fact that you need to handle error output now

**Pull request closed without merging**